packages/snaps-controllers/src/websocket/WebSocketService.ts

```diff
@@ -284,8 +286,6 @@ export class WebSocketService {
     const { socket } = this.#get(snapId, id);
 
     socket.close();
-
-    this.#sockets.delete(id);
```

Comment:
Why not call #close from the `close` event handler instead?

Reply:
Potentially, it seems simpler to manage in the event handler though. I.e. we don't need to introduce complexity in the #close function